src/ClientData/include/ClientData/ProcessData.h

```diff
@@ -90,6 +90,9 @@ class ProcessData final {
   [[nodiscard]] std::vector<std::string> FindModuleBuildIdsByPath(
       const std::string& module_path) const;
 
+  [[nodiscard]] std::vector<ModuleInMemory> FindModulesByFilename(
```

Comment:
Of course the function name already says it, but a little comment reiterating how the retrieval is exclusively by name and not by path could be useful to prevent confusion. I'm saying this because the first thing I thought was "let me look at the implementation and see if filename actually means filename or the full path".